compiler/util/clangUtil.cpp

```diff
@@ -1571,7 +1577,15 @@ void runClang(const char* just_parse_filename) {
       CHPL_TARGET_BACKEND_ARCH[0] != '\0' &&
       0 != strcmp(CHPL_TARGET_BACKEND_ARCH, "none")) {
     std::string march = "-march=";
-    march += CHPL_TARGET_BACKEND_ARCH;
+    const char *backend_arch = CHPL_TARGET_BACKEND_ARCH;
+    if (strncmp(backend_arch, "arm-", 4) == 0) {
+      backend_arch += 4;
+      march = "-mcpu=";
+    }
+    march += backend_arch;
+    if (strcmp(backend_arch, "thunderx2") == 0) {
+      march += "t99";
+    }
```

Comment:
Shouldn't this happen in util/chplenv/chpl_arch.py ? If you change it much from what you've tested here, let's put it in a different PR.

Reply:
OK, I'll leave this here for now, but we can think about whether this should migrate to chpl_arch.py in the future.  If we migrate it, it gets more complicated.

The answer we would want from chpl_arch.py depends on whether we are compiling with the `--llvm` flag.  If so, we need to translate from Cray module syntax to Clang syntax, but if not, we need to keep the Cray module syntax so the PrgEnv compiler can find the architecture properly.  If we were to put it in chpl_arch.py, we would have to default to the PrgEnv-style answer, and have the Clang driver run the Python program with a special command line argument to say that we want the Clang-style answer, and then wait for the response, or something like that.

The complexity of migrating it to chpl_arch.py might decrease if at some time in the future, compiling with LLVM means we always use LLVM and never have to worry about whether or not we have used the `--llvm` flag.

Reply:
chpl_arch.py / chpl_compiler.py already works that way. Search for CHPL_LLVM_CODEGEN in the util/chplenv scripts to see more. That environment variable is set when `--llvm` is used.

Reply:
OK, thanks.  Sorry I missed that before.  I looked for several things, including chpl_arch.py, but I didn't think to look for printchplenv being called from the driver.

Now that I see this, I agree that in the long run, this should be migrated to chpl_arch.py.  That will require a rework of chpl_arch.py to handle the difference between x86-64, where `-march=` is needed, and ARM, where `mcpu=` is needed instead.  Therefore, migrating it to chpl_arch.py does belong in a separate PR.